src/uint/boxed/shl.rs

```diff
@@ -10,27 +10,37 @@ impl BoxedUint {
     /// Returns a zero and a truthy `Choice` if `shift >= self.bits_precision()`,
     /// or the result and a falsy `Choice` otherwise.
     pub fn shl(&self, shift: u32) -> (Self, Choice) {
```

Comment:
Perhaps this should be `shl_with_carry` so as to not have a different type signature from `Shl::shl`?